src/building-blocks/FormItem.tsx

```diff
@@ -69,9 +69,31 @@ class FormItem extends Component<IFormFieldProps> {
     };
   }
 
+  private get getFieldConfig () {
+    const { fieldConfig, formModel } = this.props
+      , { field, formItemRenderExtra } = fieldConfig
+      , extraValue = get(formModel, field)
+      ;
+
+    if (!extraValue) { return fieldConfig; }
```

Comment:
There are two instances where you return `fieldConfig`. You can combine them with some boolean logic